iroh-gateway/src/error.rs

```diff
@@ -45,12 +49,19 @@ impl IntoResponse for GatewayError {
                 rb.body(BoxBody::default()).unwrap()
             }
             _ => {
-                let body = axum::Json(json!({
-                    "code": self.status_code.as_u16(),
-                    "success": false,
-                    "message": self.message,
-                    "trace_id": self.trace_id,
-                }));
+                let body = match has_trace_id {
```

Comment:
Why a match instead of a if? Does clippy allow that?

Reply:
Now that I look at it it is a bit odd for a boolean flag (was something else before though) and idk I just got used to the `match` magic :)

But yeah clippy is happy with it. 📎

Reply:
It's not wrong, and also not difficult to read, just unusual. So 🤷‍♂️